src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java

```diff
@@ -95,6 +98,20 @@ public void set(BinlogPosition p) {
 		position.set(p);
 	}
 
+	public void setSync(BinlogPosition p) {
+		LOGGER.debug("syncing binlog position: " + p);
+		position.set(p);
+		while ( true ) {
+			thread.interrupt();
+			BinlogPosition s = storedPosition.get();
+			if ( p.newerThan(s) ) {
```

Comment:
i think we could theoretically miss the bit where "our" position is stored?  ie another thread comes along and stores a newer position than ours before we get a chance to call `storedPosition.get`.

probably unlikely in practice, though.